### PR TITLE
Coverity findings

### PR DESCRIPTION
**Pull request by tjs-intel**

### CODE REVIEW

File: src/python/zigopt/authentication/login.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def right_code(user: User, code: str) -> bool:
     user.email_verification_timestamp
     and (now - user.email_verification_timestamp) < VERIFICATION_THRESHOLD
     and user.hashed_email_verification_code
-    and password_matches(code, user.hashed_email_verification_code or "")
+    and password_matches(code, user.hashed_email_verification_code)
```
Comment (dwanderson-intel) [inline src/python/zigopt/authentication/login.py:29]: is the default for a "non"code `None` or `""`; if the former, will this still run correctly or spew an error?

**Review by dwanderson-intel — APPROVED** [review]
LGTM, let's just confirm the auth-login won't throw unexpected errors before merging.

File: src/python/zigopt/authentication/login.py (hunk 1 above)
Comment (tjs-intel) [reply src/python/zigopt/authentication/login.py:29]: The error from coverity was pointing out that in the following statement:
```python
user.hashed_email_verification_code
    and password_matches(code, user.hashed_email_verification_code or "")
```
if the `password_matches` function is reached then `user.hashed_email_verification_code` is necessarily truthy, therefore the fallback `""` will never be used.

**Comment by tjs-intel** [comment]
@dwanderson-intel I plan to open a new PR for each of these changes so I can isolate test errors. Ideally I don't have to merge this one because the diff will have no changes.

**Pull request closed without merging**